secantTests/Util/RecoveryPhraseStorageTests.swift

```diff
@@ -0,0 +1,228 @@
+                    XCTAssertEqual(birthdaySet, birthday.intValue, "Keychain: birthday stored and retrieved must be the same.")
+                } else {
+                    XCTFail("Keychain: `testBirthdayStoredSucessfuly` birthday decoding failed.")
+                }
+            } catch {
+                XCTFail("Keychain: `testBirthdayStoredSucessfuly` birthday decoding failed.")
+            }
+        } catch let err {
+            XCTFail("Keychain: no error is expected for `testBirthdayStoredSucessfuly` but received. \(err)")
+        }
+    }
+
+    func testBirthdayDuplicate() throws {
+        do {
+            try storage?.importBirthday(birthdaySet)
+            try storage?.importBirthday(birthdaySet)
+            
+            XCTFail("Keychain: `testBirthdayDuplicate` is expected to throw a `duplicate` error but passed instead.")
+        } catch {
+            guard let error = error as? KeyStoringError else {
+                XCTFail("Keychain: the error is expected to be KeyStoringError but it's \(error).")
+                
+                return
+            }
+            
+            XCTAssertEqual(
+                error.debugValue,
+                KeyStoringError.alreadyImported.debugValue,
+                "Keychain: error must be .alreadyImported but it's \(error)."
+            )
+        }
+    }
+
+    func testBirthdayNoDataFound() throws {
+        do {
+            _ = try storage?.exportBirthday()
+            
+            XCTFail("Keychain: `testBirthdayNoDataFound` should fail but received some data.")
+        } catch {
+            guard let error = error as? KeyStoringError else {
+                return XCTFail("Keychain: the error is expected to be KeyStoringError but it's \(error).")
+            }
+
+            XCTAssertEqual(error.debugValue, KeyStoringError.uninitializedWallet.debugValue, "Keychain: error must be .noDataFound")
+        }
+    }
+    
+    func testDeleteBirthday() throws {
+        do {
+            // set the data (do not rely on RecoveryPhraseStorage)
+            do {
+                let data = try NSKeyedArchiver.archivedData(withRootObject: NSNumber(value: birthdaySet), requiringSecureCoding: true)
+                
+                try setData(data, forKey: "zcashBirthday")
+            } catch {
+                XCTFail("`testDeleteBirthday` storing `birthdaySet` failed.")
+            }
+
+            // test deletion of the birthday
+            storage?.nukeBirthday()
+            
+            // check presence of the key/data
+            let data = data(forKey: "zcashBirthday")
+            
+            XCTAssertEqual(data, nil, "Keychain: keychain is expected to not find anything for key `zcashBirthday` but received some data.")
+        }
+    }
+    
+    // MARK: - Misc
+    
+    /// The followings methods are here purposely to not rely on `RecoveryPhraseStorage` in order to test functionality of JUST ONE method at a time
```

Comment:
Since you are PRAGMA marking these and commenting about the usage of the entire group, you could also do a private extension here, it might be easier to read.

Reply:
done